src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/hearing/HearingFormGenerator.java

```diff
@@ -0,0 +1,124 @@
+            documentGeneratorService.generateDocmosisDocument(templateData, template);
+        caseDocuments.add(documentManagementService.uploadDocument(
+            authorisation,
+            new PDF(
+                getFileName(caseData, template),
+                document.getBytes(),
+                DocumentType.HEARING_FORM
+            )
+        ));
+        return caseDocuments;
+    }
+
+    @Override
+    public HearingForm getTemplateData(CaseData caseData) {
+
+        return HearingForm.builder()
+            .court(caseData.getHearingLocation().getValue().getLabel())
+            .caseNumber(caseData.getLegacyCaseReference())
+            .creationDate(getDateFormatted(LocalDate.now()))
+            .claimant(caseData.getApplicant1().getPartyName())
+            .claimantReference(checkReference(caseData)
+                                   ? caseData.getSolicitorReferences().getApplicantSolicitor1Reference() : null)
+            .defendant(caseData.getRespondent1().getPartyName())
+            .defendantReference(checkReference(caseData)
+                                    ? caseData.getSolicitorReferences().getRespondentSolicitor1Reference() : null)
+            .hearingDate(getDateFormatted(caseData.getHearingDate()))
+            .hearingTime(getHearingTimeFormatted(caseData.getHearingTimeHourMinute()))
+            .hearingType(getHearingType(caseData))
+            .applicationDate(getDateFormatted(caseData.getDateOfApplication()))
+            .hearingDuration(getHearingDuration(caseData))
+            .additionalInfo(caseData.getInformation())
+            .feeAmount(HearingUtils.formatHearingFee(caseData.getHearingFee()))
+            .hearingDueDate(getDateFormatted(caseData.getHearingDueDate()))
+            .additionalText(caseData.getHearingNoticeListOther())
+            .claimant2exists(nonNull(caseData.getApplicant2()))
+            .defendant2exists(nonNull(caseData.getApplicant2()))
+            .claimant2(nonNull(caseData.getApplicant2()) ? caseData.getApplicant2().getPartyName() : null)
+            .defendant2(nonNull(caseData.getApplicant2()) ? caseData.getRespondent2().getPartyName() : null)
+            .claimant2Reference(checkReference(caseData)
+                                    ? caseData.getSolicitorReferences().getApplicantSolicitor1Reference() : null)
+            .defendant2Reference(checkReference(caseData)
+                                     ? caseData.getSolicitorReferences().getRespondentSolicitor2Reference() : null)
+            .build();
+    }
+
+    private String getFileName(CaseData caseData, DocmosisTemplates template) {
+        return String.format(template.getDocumentTitle(), caseData.getLegacyCaseReference());
+    }
+
+    private String getHearingTimeFormatted(String hearingTime) {
+        StringBuilder hearingTimeBuilder = new StringBuilder(hearingTime);
+        hearingTimeBuilder.insert(2, ':');
+        return hearingTimeBuilder.toString();
```

Comment:
This works on the condition that `hearingTime` is a string exactly 4 (or 7, if we use AM/PM notation) characters long.
I normally don't trust input data. I'd recommend to add a validation on the `hearingTime` string length, to ensure it's exactly as long as we expect it to be.

Also: is the time in 24h or 12h format? Are we formatting in the same 12/24 format or we want to ensure one over the other?

Given all the above I'd like to have a unit test on this function, to try all the various allowed and disallowed inputs and make sure the output is always consistent.

Reply:
What is formatted are the values of a FixedList. They are strings with the format "0800", "0830"...The format is the same as the one received from ccd. I have added a test to check the format